### PR TITLE
Repl enhancements 2

### PR DESCRIPTION
**Pull request by dhoomakethu**
* Fix `--random` in `pymodbus.server` to randomize values read for `discrete inputs` and `input registers`
image

Example invocation
```
pymodbus.server --verbose run  --modbus-server tcp --modbus-framer socket --modbus-port 5020 --unit-id 1 --unit-id 2 -u 4 -r 1 --timeout 2
```

The values for simulation can also be passed as a json file to `--modbus-config` param , for e.g to limit the max and min values a register should have. Refer `data_block_settings` in the below json file.

```
{
  "tcp": {
        "handler": "ModbusConnectedRequestHandler",
        "allow_reuse_address": true,
        "allow_reuse_port": true,
        "backlog": 20,
        "ignore_missing_slaves": false
    },
    "serial": {
        "handler": "ModbusSingleRequestHandler",
        "stopbits": 1,
        "bytesize": 8,
        "parity": "N",
        "baudrate": 9600,
        "timeout": 3,
        "auto_reconnect": false,
        "reconnect_delay": 2
    },
    "tls": {
        "handler": "ModbusConnectedRequestHandler",
        "certfile": null,
        "keyfile": null,
        "allow_reuse_address": true,
        "allow_reuse_port": true,
        "backlog": 20,
        "ignore_missing_slaves": false
    },
    "udp": {
        "handler": "ModbusDisonnectedRequestHandler",
        "ignore_missing_slaves": false
    },
    "data_block_settings": {
        "min_binary_value": 0,
        "max_binary_value": 1,
        "min_register_value": 0,
        "max_register_value": 100,
        "data_block": {
            "discrete_inputs": {
                "block_start": 0,
                "block_size": 100,
                "default": 0,
                "sparse": false
            },
            "coils": {
                "block_start": 0,
                "block_size": 100,
                "default": 0,
                "sparse": false
            },
            "holding_registers": {
                "block_start": 0,
                "block_size": 100,
                "default": 0,
                "sparse": false
            },
            "input_registers": {
                "block_start": 0,
                "block_size": 100,
                "default": 0,
                "sparse": false
            }

        }


    }
}
```

# TODO

- [ ] Update unit tests
- [x] Update documentation

### CODE REVIEW

**Comment by dhoomakethu** [comment]
Updated documentation https://github.com/riptideio/pymodbus/blob/repl-enhancements-2/pymodbus/repl/server/README.md

**Comment by janiversen** [comment]
I am hoping to make 3.0.1 around November 5, does that give you enough time to merge this ?

If not I will pospone the release a bit.

**Comment by janiversen** [comment]
The CI error might be mine, will look at it Monday.

**Comment by dhoomakethu** [comment]
Ok, I will poke around a bit tomorrow to see if I can fix it.

**Comment by janiversen** [comment]
Don’t’t worry as you can see it is a temporary error, restarting the job cures the problem.

File: pymodbus/repl/server/README.md [hunk 1]
```diff
@@ -120,3 +120,104 @@ Usage: manipulator response_type=|normal|error|delayed|empty|stray
...
+
+      ```
+  * `delayed`: Responses are delayed by the time specified with `delay_by` field.
+    * `delay_by`: Delay the response by <n> seconds (`float`).
+  * `empty`: Returns an empty response or no response
+  * `stray`: Returns stray/junk response with length specified with `data_len` field.
+    * `data_len`: Length of the stray_data (`int`)
+  * `clear_after`: Clears the error responses after <n> requests (`int`)
+
+**EXAMPLES**
```
Comment (serroba) [inline pymodbus/repl/server/README.md:186]: thanks a lot for adding this example section. Super useful

File: pymodbus/server/reactive/main.py [hunk 2]
```diff
@@ -92,6 +102,85 @@
...
+        max_register_value = kwargs.get("max_register_value", 65535)
+        self._randomize = randomize
+        self._lock = threading.Lock()
+        self._read_counter = {"d": 0, "i": 0}
+        self._min_binary_value = min_binary_value
+        self._max_binary_value = max_binary_value
+        self._min_register_value = min_register_value
+        self._max_register_value = max_register_value
+
+    def getValues(self, fc_as_hex, address, count=1):
```
Comment (serroba) [inline pymodbus/server/reactive/main.py:152]: should this functionality work for single bit object types? I was thinking about coils.
Not sure if is really much value, but for my own tests that would be nice, although the complexity of randomise booleans is much less.

Anyhow, just wondering about your thoughts about those objects.

Comment (dhoomakethu) [reply pymodbus/server/reactive/main.py:152]: Holding registers and coils can be overridden and hence they are excluded from simulation. Bit fields should be straight forward (0 or 1)

Comment (serroba) [reply pymodbus/server/reactive/main.py:152]: Right that makes sense

**Comment by dhoomakethu** [comment]
@janiversen I may not be able to finish writing the test cases for this PR by first week of November, Please feel free to drop this from the next release if you think having tests are a must. You can merge otherwise and I will add tests in future.

**Comment by janiversen** [comment]
I will merge this now, because I am testing/pre-building doc and sources to see what more we need, please make the tests in a new PR.

**Pull request merged**